pkg/agent/flowexporter/connections/conntrack_connections_perf_test.go

```diff
@@ -0,0 +1,175 @@
+*/
+
+func BenchmarkPoll(b *testing.B) {
+	disableLogToStderr()
+	connStore, mockConnDumper := setupConntrackConnStore(b)
+	conns := generateConns()
+	for n := 0; n < b.N; n++ {
+		mockConnDumper.EXPECT().DumpFlows(uint16(openflow.CtZone)).Return(conns, testNumOfConns, nil)
+		connStore.Poll()
+		b.StopTimer()
+		conns = generateUpdatedConns(conns)
+		b.StartTimer()
+	}
+	b.Logf("\nSummary:\nNumber of initial connections: %d\nNumber of new connections/poll: %d\nNumber of deleted connections/poll: %d\n", testNumOfConns, testNumOfNewConns, testNumOfDeletedConns)
+}
+
+func setupConntrackConnStore(b *testing.B) (*ConntrackConnectionStore, *connectionstest.MockConnTrackDumper) {
+	ctrl := gomock.NewController(b)
+	defer ctrl.Finish()
+	mockIfaceStore := interfacestoretest.NewMockInterfaceStore(ctrl)
+	testInterface := &interfacestore.InterfaceConfig{
+		Type: interfacestore.ContainerInterface,
+		ContainerInterfaceConfig: &interfacestore.ContainerInterfaceConfig{
+			PodName:      "pod",
+			PodNamespace: "pod-ns",
+		},
+	}
+	mockIfaceStore.EXPECT().GetInterfaceByIP(gomock.Any()).Return(testInterface, true).AnyTimes()
+
+	mockConnDumper := connectionstest.NewMockConnTrackDumper(ctrl)
+	mockConnDumper.EXPECT().GetMaxConnections().Return(100000, nil).AnyTimes()
+
+	serviceStr := "10.0.0.1:30000/TCP"
+	servicePortName := k8sproxy.ServicePortName{
+		NamespacedName: types.NamespacedName{
+			Namespace: "serviceNS1",
+			Name:      "service1",
+		},
+		Port:     "30000",
+		Protocol: v1.ProtocolTCP,
+	}
+	mockProxier := proxytest.NewMockProxier(ctrl)
+	mockProxier.EXPECT().GetServiceByIP(serviceStr).Return(servicePortName, true).AnyTimes()
+
+	npQuerier := queriertest.NewMockAgentNetworkPolicyInfoQuerier(ctrl)
+
+	return NewConntrackConnectionStore(mockConnDumper, flowrecords.NewFlowRecords(), mockIfaceStore, true, false, mockProxier, npQuerier, testPollInterval), mockConnDumper
+}
+
+func generateConns() []*flowexporter.Connection {
+	conns := make([]*flowexporter.Connection, testNumOfConns)
+	for i := 0; i < testNumOfConns; i++ {
+		conns[i] = getNewConn()
+	}
+	return conns
+}
+
+func generateUpdatedConns(conns []*flowexporter.Connection) []*flowexporter.Connection {
+	length := len(conns) - testNumOfDeletedConns + testNumOfNewConns
+	updatedConns := make([]*flowexporter.Connection, length)
+	for i := 0; i < len(conns); i++ {
+		// replace deleted connection with new connection
+		if conns[i].DoneExport == true {
+			conns[i] = getNewConn()
+		} else { // update rest of connections
+			conns[i].OriginalPackets += 5
+			conns[i].OriginalBytes += 20
+			conns[i].ReversePackets += 2
+			conns[i].ReverseBytes += 10
+		}
+		updatedConns[i] = conns[i]
+	}
+	for i := len(conns); i < length; i++ {
+		updatedConns[i] = getNewConn()
+	}
+	randomNum := getRandomNum(int64(length - testNumOfDeletedConns))
+	for i := randomNum; i < testNumOfDeletedConns+randomNum; i++ {
+		updatedConns[i].DoneExport = true
```

Comment:
what if `updatedConns[i]` is one of the new connections for this iteration? Does it still make sense?

Reply:
In my opinion it is still valid because the connection can be stored and will be deleted in the next round. What do you think?

Reply:
I think it's ok since we are not testing export here. It would have been good to add a comment to this effect.